### DISCUSSION TITLE
Two Form fields with the same name

### DISCUSSION THREAD

**Discussion by PseudoDevs** [filamentphp/filament#9631] [Help] [answered]
Hi everyone,

I'm new to Filament. Based on the form fields that I've created, the File and TextInput fields depend on the value of the Select field.

It works fine for hiding the fields if they weren't selected. Additionally, it works for inserting the "File" upload field. However, when I need the TextInput for the URL, it doesn't return as expected.

What is the best/better practice to solve this?

Thank you so much.

![image](https://github.com/filamentphp/filament/assets/28098260/930ce44e-0e4e-41c1-89e2-392154d91bde)

**FORM FIELDS:**
![image](https://github.com/filamentphp/filament/assets/28098260/285c43ec-3338-4da9-a05b-3b752ef5def1)

**Comment by sinnbeck** [comment]
Give them different names and use `formatStateUsing()` to set the data on the text input. I cannot copy your code from an image so I cannot give and example.

**Reply by PseudoDevs** [reply]
Good day, Sorry for the late reply, 
Here's my code from the image above, 
Still can't figure it out how to make this work also in the edit form


Thank you, :) 

```php
        Select::make('file_type')
                                ->native(false)
                                ->searchable()
                                ->live()
                                ->options([
                                    'file' => 'File',
                                    'url' => 'URL',
                                ])->afterStateUpdated(function (Get $get, Set $set, ?string $old, ?string $state) {
                                    $set('type', $state);
                                }),
    
                            // Show FileUpload if file_type is file and TextInput if file_type is url
    
                            FileUpload::make('filename')
                            ->label('Upload File')
                            ->acceptedFileTypes(['application/pdf', 'application/csv', 'application/xlsx', 'application/docx', 'application/pptx'])
                            ->directory('files')
                            ->formatStateUsing(function (File $file) {
                                return $file ? Str::after($file, 'files/') : null;
                            })
                            ->visible(fn (Get $get) => $get('file_type') === 'file'),
    
                            TextInput::make('filename')
                            ->label('File URL')
                            ->url()
                            ->visible(fn (Get $get) => $get('file_type') === 'url'),
```

**Comment by PseudoDevs** [comment, marked as answer]
I have already solved it using ->statePath('column_name'), Thank you helping me :)

**Reply by kombiahrk** [reply]
can you share the code it will be helpful

**Reply by hardiklakhalani** [reply]
@kombiahrk  This worked for me
statePath kind of works as an alias.
```php
Select::make('file_type')
       ->statePath('file_type_profile')
       ...
```
```php
Select::make('file_type')
       ->statePath('file_type_xyz')
       ...
```
